metrics/segmentStats.py

```diff
@@ -0,0 +1,506 @@
+            source node, target node, value is the contraction of the segment (0) since it is a cycle
+        """
+        sourceOnCycle = cycle[0]
+        if sourceOnCycle not in self._visitedSources:
+            self.countDict[sourceOnCycle] = 1
+            self._visitedSources.append(sourceOnCycle)
+        else:
+            self.countDict[sourceOnCycle] += 1
+        curveLength = self._getLengthAndRemoveTracedPath(subGraphSkeleton, cycle, 1)
+        self.cycleInfoDict[self.cycles] = [0, curveLength]
+        self.lengthDict[self.countDict[sourceOnCycle], sourceOnCycle, cycle[len(cycle) - 1]] = curveLength
+        self.tortuosityDict[self.countDict[sourceOnCycle], sourceOnCycle, cycle[len(cycle) - 1]] = 0
+        self.contractionDict[self.countDict[sourceOnCycle], sourceOnCycle, cycle[len(cycle) - 1]] = 0
+
+    def _cyclicTree(self, subGraphSkeleton, cycleList):
+        """
+        Find statistics of a cyclic tree of a disjoint graph
+
+        Parameters
+        ----------
+        subGraphSkeleton: Networkx Graph
+            disjoint subgraph of the skeleton's networkx graph
+
+        cyclesList: list of lists
+            list of list of nodes in the cycles
+
+        cycles: int
+            number of cycles in the graph so far
+        Returns
+        -------
+        subGraphSkeleton : Networkx graph
+            graph changed inplace with visited nodes removed
+
+        cycleInfoDict : Dictionary
+            class instance variable changed in place with key as nth cycle, value as list with
+            [number of branch points on cycle, length of the cycle]
+
+        countDict : Dictionary
+            class instance variable changed in place with key as source node,
+            value is the number of segments attached to the source node
+
+        lengthDict : Dictionary
+            class instance variable changed in place with key as index of the segment from the source node,
+            source node, target node, value is the length of the segment
+
+        tortuosityDict : Dictionary
+            class instance variable changed in place with key as index of the segment from the source node,
+            source node, target node, value is the tortuosity of the segment
+
+        contractionDictdict : Dictionary
+            class instance variable changed in place with key as index of the segment from the source node,
+            source node, target node, value is the contraction of the segment
+
+        hausdorffDimensionDict : Dictionary
+            class instance variable changed in place with key as index of the segment from the source node,
+            source node, target node, value is the hausdorff dimension of the segment
+        """
+        for nthcycle, cycle in enumerate(cycleList):
+            nodeDegreeDictFilt = {key: value for key, value in self._nodeDegreedict.items() if key in cycle}
+            branchPointsOnCycle = [k for (k, v) in nodeDegreeDictFilt.items() if v != 2 and v != 1]
```

Comment:
I'm not entirely clear on why we're filtering out 1 and 2. Are we really filtering out anything > 0, or < 3? I'm confused about this line.